### ISSUE TITLE
Ubuntu 20.04 support

### ISSUE THREAD

**Issue by ignl** [Studio3T/robomongo#1726]
Hi, after upgrading server to Ubuntu 20.04 ssh tunnel stopped working.

**Comment by harshmandan** [comment]
+1 Updated my server to 20.04 yesterday and wasted 2 hours trying different firewalls configs and even different software. Can't connect to my server.

**Comment by maikeldotuk** [comment]
Same problem.

**Comment by pitpitpat** [comment]
Same here. Spinned up a fresh VM with Ubuntu 20.04 and SSH tunnel could not connect.

Error Details:
> Failed to create SSH tunnel to 104.248.21.193:20.
> 
> Error:
> Operation timed out. Failed to connect to 104.248.21.193:20. (Error #60)

**Comment by maikeldotuk** [comment]
Hi @pitpitpat , my solution was to scrap that machine and install instead Ubuntu 18 on it. For some reason, which I think has to do with the OpenSSH version, it work flawlessly there. So for now, my solution has been to downgrade ubuntu.

**Comment by simsekgokhan** [comment]
Hi @ignl and all, thanks a lot for reporting the problem. 
We did reproduce the problem. And it is reproducible even with the currently developed Robo 1.4 which uses latest openssl-1.1.1f and latest libssh2 1.9.0. We will have to investigate more to understand the root cause of this problem.

**Comment by simsekgokhan** [comment]
We have done little bit more testing. SSH connection worked after adding public key into server `authorized_keys` file. If someone wants to try, this solved it:

```
// SSH server:  
~/.ssh/authorized_keys // public keys of clients this server trusts, one line per key
	ssh-rsa <client-pub-key>
```
Important: Currently DSA, RSA keys are supported. Robo 1.4 is being develped to support ECDSA and Ed25519 keys.

**Comment by maikeldotuk** [comment]
Sounds interesting, will give it a try tonight (Spain time) On 13 Jul 2020 16:45, Gökhan Şimşek <notifications@github.com> wrote:
We have done little bit more testing. SSH connection worked after adding public key into server authorized_keys file. If someone wants to try, this solved it:
// SSH server:  
~/.ssh/authorized_keys // public keys of clients this server trusts, one line per key
	ssh-rsa <client-pub-key>

Important: Currently DSA, RSA keys are supported. Robo 1.4 is being develped to support ECDSA and Ed25519 keys.

**Comment by harshmandan** [comment]
Waiting for ecda key support eagerly!

**Comment by pitpitpat** [comment]
> We have done a little bit more testing with Ubuntu 20.04. SSH connection worked after adding public key into server `authorized_keys` file. If someone wants to try, this solved it:
> 
> ```
> // SSH server:  
> ~/.ssh/authorized_keys // public keys of clients this server trusts, one line per key
> ssh-rsa <client-pub-key>
> ```
> 
> Important: Currently DSA, RSA keys are supported. Robo 1.4 is being develped to support ECDSA and Ed25519 keys. PPK keys must be converted to OPENSSH format.

@simsekgokhan 
The issue seems to persist when using SSH tunnel option, even after adding the public RSA key to the server's `authorized_keys` file.
Tested on MacOS 10.15.5.
Server is running Ubuntu 20.04 (on DigitalOcean)

**Comment by maikeldotuk** [comment]
Right, so RSA key was already on authorised keys on test server with Ubuntu 20.04, the proposed solution same as for @pitpitpat didn't work. The problem still exists.

**Comment by harshmandan** [comment]
@maikeldotuk Any update on this?

**Comment by maikeldotuk** [comment]
It still doesn't work for me. On another system I can't afford to downgrade Ubuntu, I've tried everything and robo3t can't connect while nosqlbooster simply connects with no issue.
________________________________

**Comment by maikeldotuk** [comment]
So for that system I'm using nosqlbooster. I have no other option.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by simsekgokhan** [comment]
Please also see: https://github.com/Studio3T/robomongo/issues/1189#issuecomment-447109457

**Comment by ufoproger** [comment]
Step-by-step solution based on https://github.com/Studio3T/robomongo/issues/1189#issuecomment-750128888.

Tested for OpenSSH server 8.3+ (installed on Ubuntu 20.04 and newer, Arch Linux etc).

You need to add legacy key exchange algorithm `diffie-hellman-group14-sha1` to OpenSSH server (parameter `KexAlgorithms`):

1. Get default value of parameter `KexAlgorithms`: run `man 5 sshd_config` ([example](https://man.openbsd.org/sshd_config#KexAlgorithms)):
```
KexAlgorithms
	...

	The default is:

		curve25519-sha256,curve25519-sha256@libssh.org,
		ecdh-sha2-nistp256,ecdh-sha2-nistp384,ecdh-sha2-nistp521,
		diffie-hellman-group-exchange-sha256,
		diffie-hellman-group16-sha512,diffie-hellman-group18-sha512,
		diffie-hellman-group14-sha256
```
2. The current value of the parameter `KexAlgorithms` is `curve25519-sha256,curve25519-sha256@libssh.org,ecdh-sha2-nistp256,ecdh-sha2-nistp384,ecdh-sha2-nistp521,diffie-hellman-group-exchange-sha256,diffie-hellman-group16-sha512,diffie-hellman-group18-sha512,diffie-hellman-group14-sha256`;
3. Append legacy kex algorithm `diffie-hellman-group14-sha1` to the parameter from above;
4. Add the resulting parameter to the end of the file `/etc/ssh/sshd_config`:
```
KexAlgorithms curve25519-sha256,curve25519-sha256@libssh.org,ecdh-sha2-nistp256,ecdh-sha2-nistp384,ecdh-sha2-nistp521,diffie-hellman-group-exchange-sha256,diffie-hellman-group16-sha512,diffie-hellman-group18-sha512,diffie-hellman-group14-sha256,diffie-hellman-group14-sha1
```
5. Test configuration: `sshd -t`;
6. Reload OpenSSH server: `systemctl reload sshd` or `service sshd reload` etc;
7. enjoy using the built-in Robo 3T SSH tunnel :)

**Comment by Rush** [comment]
Unfortunately the workaround didn't work for me. It looks like robo3t is abandoned which also does not give me any reason to try Studio 3T. Are there any good alternatives?

**Comment by melkamar** [comment]
I had this exact error after I have upgraded the server to 20.04. Upgrading the Robo 3T client to latest `Robo 3T 1.4.3 (Build 48f7dfd - 8/2021)` fixed it for me and I can connect again.

**Comment by dns2utf8** [comment]
Robo 3T uses libssh2 version 1.7.0 from 2016 <- 5 years ago.
An update to 1.10.0 from last summer would fix the issue without any workaround

https://www.libssh2.org/changes.html
